lib/src/utilities/configuration.dart

```diff
@@ -0,0 +1,38 @@
+/// Copyright (C) 2021-2022 Intel Corporation
+/// SPDX-License-Identifier: BSD-3-Clause
+///
+/// configuration.dart
+/// Extract configuration of ROHD from pubspec YAML
+///
+/// 2021 May 7
+/// Author: Yao Jing Quek <yao.jing.quek@intel.com>
+///
+
+import 'dart:io';
+import 'package:yaml/yaml.dart';
+
+/// A utility to extract the ROHD configuration from pubspec.yaml
+/// configuration document.
+abstract class Configuration {
+  /// A getter to return the current configuration of ROHD
+  static Map<String, dynamic> get getConfig {
+    const config = './pubspec.yaml';
+    final f = File(config);
+    final yamlText = f.readAsStringSync();
+    final yaml = loadYaml(yamlText) as Map;
+
```

Comment:
Change to String and return only version. How do you get the path to ROHD when you not working on rohd? They have their own ``pubspec.yaml`` instead? 

Can use dart tool to define the path? Need to query for correct location where the version is. 

Check out this:
``https://pub.dev/packages/package_config``

Easy but not elegant
Make another dart class with the update of the version. Add a CI check if the version in there not match with ``pubspec.yaml`` then fail. 

Or:
look from ``package.json`` but its a hacky method.


## Git Hash
- Add https://dart.dev/tools/pub/publishing 
- Make a script wrapper around publish instead run it manually

Reply:
https://suragch.medium.com/reading-configuration-files-and-settings-in-flutter-and-dart-cc14bbd93698